### PR TITLE
Mempool: handle not-executable transactions related to guardians

### PR DESCRIPTION
**Pull request by andreibancioiu**
## Reasoning behind the pull request
- This is an improvement upon https://github.com/multiversx/mx-chain-go/pull/6603.
- We should have no / as little as possible not-executable transactions in the bulk of selected transactions.
  
## Proposed changes
- https://github.com/multiversx/mx-chain-storage-go/pull/58
- Inform selection about not-executable transactions related to guardians, through exporting & using the `VerifyGuardian` function.

## Testing procedure
- Standard testing
- Send not-executable transactions (related to guardians). Inspect the logs and assert they aren't selected.

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: process/transaction/baseProcess.go [hunk 1]
```diff
@@ -288,7 +288,7 @@ func (txProc *baseTxProcessor) checkGuardedAccountUnguardedTxPermission(tx *tran
 	return nil
 }
 
-func (txProc *baseTxProcessor) verifyGuardian(tx *transaction.Transaction, account state.UserAccountHandler) error {
+func (txProc *baseTxProcessor) VerifyGuardian(tx *transaction.Transaction, account state.UserAccountHandler) error {
```
Comment (andreibancioiu) [inline process/transaction/baseProcess.go:291]: We not export this function (design workaround).

File: process/block/preprocess/selectionSession.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		Nonce:   userAccount.GetNonce(),
+		Balance: userAccount.GetBalance(),
+	}, nil
+}
+
+// IsBadlyGuarded checks if a transaction is badly guarded (not executable).
+// Will be called by mempool during transaction selection.
+func (session *selectionSession) IsBadlyGuarded(tx data.TransactionHandler) bool {
+	address := tx.GetSndAddr()
+	account, err := session.accountsAdapter.GetExistingAccount(address)
```
Comment (andreibancioiu) [inline process/block/preprocess/selectionSession.go:56]: In the near future, we'll add a memoization mechanism here (at a higher level than the accounts adapter cache).

File: process/block/preprocess/selectionSession.go [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+	if err != nil {
+		return false
+	}
+
+	userAccount, ok := account.(state.UserAccountHandler)
```
Comment (raduchis) [inline process/block/preprocess/selectionSession.go:61]: a similar check is done on the VerifyGuardian an that returns the `ErrTransactionNotExecutable` error 

https://github.com/multiversx/mx-chain-go/blob/4c24359ab89654bee5b184d2506d31cd959165bf/process/transaction/baseProcess.go#L307

File: process/transaction/baseProcess.go (hunk 1 above)
Comment (raduchis) [reply process/transaction/baseProcess.go:291]: We not?

File: process/transaction/export_test.go [hunk 4]
```diff
@@ -102,7 +102,7 @@ func (inTx *InterceptedTransaction) CheckMaxGasPrice() error {
 
 // VerifyGuardian calls the un-exported method verifyGuardian
 func (txProc *txProcessor) VerifyGuardian(tx *transaction.Transaction, account state.UserAccountHandler) error {
```
Comment (raduchis) [inline process/transaction/export_test.go:104]: there is no need for this export now

File: process/transaction/baseProcess.go (hunk 1 above)
Comment (andreibancioiu) [reply process/transaction/baseProcess.go:291]: *we now (updated message above)

File: process/transaction/export_test.go (hunk 4 above)
Comment (andreibancioiu) [reply process/transaction/export_test.go:104]: Indeed, removed (actually, by mistake, I did a bad recursive call).

File: process/block/preprocess/selectionSession.go (hunk 3 above)
Comment (andreibancioiu) [reply process/block/preprocess/selectionSession.go:61]: In the future, we have to somehow reconcile `state.UserAccountHandler` and `vmcommon.UserAccountHandler`.

Comment (raduchis) [reply process/block/preprocess/selectionSession.go:61]: ok!

**Review by raduchis — APPROVED** [review]

**Pull request merged**